### PR TITLE
Return empty Config field in FromName to avoid nil pointer error 

### PR DESCRIPTION
**Pull request by nak3**
Fix for https://github.com/openshift/origin/issues/4186

### CODE REVIEW

**Comment by bparees** [comment]
@csrwng ptal

File: pkg/generate/app/cmd/newapp.go [hunk 1]
```diff
@@ -500,7 +500,7 @@ func (c *AppConfig) buildPipelines(components app.ComponentReferences, environme
 				if ref.Input().Uses.IsDockerBuild() {
 					exposed, ok := ref.Input().Uses.Info().Dockerfile.GetDirective("EXPOSE")
 					if ok {
-						if input.Info == nil {
+						if input.Info.Config == nil {
```
Comment (bparees) [inline pkg/generate/app/cmd/newapp.go]: should this be "if input.Info!=nil && input.Info.Config!=nil" ?

Comment (nak3) [inline pkg/generate/app/cmd/newapp.go]: Yes, I think I should do `input.Info` nil check.

File: pkg/generate/app/cmd/newapp.go [hunk 2]
```diff
@@ -500,7 +500,7 @@ func (c *AppConfig) buildPipelines(components app.ComponentReferences, environme
 				if ref.Input().Uses.IsDockerBuild() {
 					exposed, ok := ref.Input().Uses.Info().Dockerfile.GetDirective("EXPOSE")
 					if ok {
-						if input.Info == nil {
+						if !(input.Info != nil && input.Info.Config != nil) {
```
Comment (bparees) [inline pkg/generate/app/cmd/newapp.go]: I don't think this is the right fix.  if input.Info is not nil but input.Info.Config is nil, you're going to clobber the input.Info object.

(also that if check could be written more simply as "if input.Info==nil || input.Info.Config==nil", but that's moot).

The right fix here is to fix the imageref.go FromName method to always populate the Config field, and/or to just populate the Config field with an empty value if the Config field is nil in the existing Info object.

**Comment by nak3** [comment]
Thank you. How about this one?

**Comment by mfojtik** [comment]
LGTM

**Comment by bparees** [comment]
lgtm [merge]

**Pull request merged**